calculator.js

```diff
@@ -0,0 +1,104 @@
+let operator = "";
+let previousValue = "";
+let currentValue = "";
+
+const choices = {
+  addition: "+",
+  subtraction: "-",
+  multiplication: "x",
+  division: "/",
+  decimal: ".",
+};
+
+document.addEventListener("DOMContentLoaded", function () {
+  const clear = document.querySelector("#clear-btn");
+  const equal = document.querySelector(".equal");
+  const decimal = document.querySelector(".decimal");
+
+  const numbers = document.querySelectorAll(".number");
+  const operators = document.querySelectorAll(".operator");
+
+  const previousScreen = document.querySelector(".previous");
+  const currentScreen = document.querySelector(".current");
+
+  numbers.forEach((number) =>
+    number.addEventListener("click", function (e) {
+      handleNumber(e.target.textContent);
+      currentScreen.textContent = currentValue;
+    })
+  );
+
+  operators.forEach((op) =>
+    op.addEventListener("click", function (e) {
+      handleOperator(e.target.textContent);
+      previousScreen.textContent = previousValue + " " + operator;
+      currentScreen.textContent = currentValue;
+    })
+  );
+
+  clear.addEventListener("click", function () {
+    previousValue = "";
+    currentValue = "";
+    operator = "";
+    previousScreen.textContent = currentValue;
+    currentScreen.textContent = currentValue;
+  });
+
+  equal.addEventListener("click", function () {
+    if (currentValue !== "" && previousValue !== "") {
+      calculate();
+      previousScreen.textContent = "";
+      if (previousValue.length <= 5) {
+        currentScreen.textContent = previousValue;
+      } else {
+        currentScreen.textContent = previousValue.slice(0, 5) + "...";
```

Comment:
instead of chopping the result, try to limit the number of decimal places using Math functions. toFixed can be one alternative, or use round, ceil, floor, trunc etc.